dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -194,7 +194,15 @@ public async Task<SKContext> RunAsync(ContextVariables variables, CancellationTo
             try
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                context = await f.InvokeAsync(context, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                if (f.IsSemantic && f is SKFunction func && !func.HasAIService())
+                {
+                    context = await f.InvokeAsync(context, textCompletion: this.GetService<ITextCompletion>(), cancellationToken: cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    context = await f.InvokeAsync(context, cancellationToken: cancellationToken).ConfigureAwait(false);
+                }
```

Comment:
I think we should investigate in more detail how to inject AI service based on future functionality like:
1. Allowing different types of AI services (not only text completion).
2. Fallback logic when first AI service is unresponsive etc.

Also, I'm wondering what was the reason of adding `ITextCompletion` parameter to `InvokeAsync` function? Because in this case we need to call `InvokeAsync` with different set of parameters, which makes it a little bit complicated, especially if the list of parameters will grow.

For example, we can inject AI service with `f.SetAIService(() => this.GetService<ITextCompletion>());` and use the same `InvokeAsync` signature which will simplify the code and remove the need to update all unit tests to add new `ITextCompletion` parameter.